tests/functional/pv/test_qos_limits.py

```
@@ -0,0 +1,568 @@
+        time.sleep(15)
+        helpers.wait_for_resource_state(pod_obj, constants.STATUS_RUNNING, timeout=420)
+
+        # 1. Verify cgroup throttle boundaries map to Gold tier
+        self.verify_node_cgroup_throttling(
+            pod_obj, pvc_obj, "kubepods-pod.slice", self.gold_limits
+        )
+
+        # 2. Verify write operation fails inside the read-only volume container via direct exec_cmd
+        log.info("Verifying write protection on ROX block volume...")
+        dd_cmd = f"oc exec {pod_obj.name} -n {proj.namespace} -- dd if=/dev/zero of=/dev/rbdblock bs=1M count=1"
+        res = exec_cmd(dd_cmd, shell=True, ignore_error=True)
```

Ruff (0.16.0)

[error] Function call with `shell=True` parameter identified, security issue

(S604)